phoenix-core/src/main/java/org/apache/phoenix/mapreduce/transform/PhoenixTransformWithViewsInputFormat.java

```diff
@@ -0,0 +1,116 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Properties;
+
+import static org.apache.phoenix.jdbc.PhoenixDatabaseMetaData.SYSTEM_CHILD_LINK_NAME_BYTES;
+import static org.apache.phoenix.mapreduce.util.PhoenixConfigurationUtil.getIndexToolIndexTableName;
+
+public class PhoenixTransformWithViewsInputFormat<T extends DBWritable> extends PhoenixServerBuildIndexInputFormat {
+    private static final Logger LOGGER =
+            LoggerFactory.getLogger(PhoenixTransformWithViewsInputFormat.class);
+    @Override
+    public List<InputSplit> getSplits(JobContext context) throws IOException, InterruptedException {
+        final Configuration configuration = context.getConfiguration();
+        try (PhoenixConnection connection = (PhoenixConnection)
+                ConnectionUtil.getInputConnection(configuration)) {
+            try (Table hTable = connection.unwrap(PhoenixConnection.class).getQueryServices().getTable(
+                    SchemaUtil.getPhysicalTableName(SYSTEM_CHILD_LINK_NAME_BYTES, configuration).toBytes())) {
+                String oldDataTableFullName = PhoenixConfigurationUtil.getIndexToolDataTableName(configuration);
+                String newDataTableFullName = getIndexToolIndexTableName(configuration);
+                PTable newDataTable = PhoenixRuntime.getTableNoCache(connection, newDataTableFullName);
+                String schemaName = SchemaUtil.getSchemaNameFromFullName(oldDataTableFullName);
+                String tableName = SchemaUtil.getTableNameFromFullName(oldDataTableFullName);
+                byte[] schemaNameBytes = Strings.isNullOrEmpty(schemaName) ? null : schemaName.getBytes();
+                Pair<List<PTable>, List<TableInfo>> allDescendantViews = ViewUtil.findAllDescendantViews(hTable, configuration, null, schemaNameBytes,
+                        tableName.getBytes(), EnvironmentEdgeManager.currentTimeMillis(), false);
+                List<PTable> legitimateDecendants = allDescendantViews.getFirst();
+
+                List<InputSplit> inputSplits = new ArrayList<>();
+
+                HashMap<String, PColumn> columnMap = new HashMap<>();
+                for (PColumn column : newDataTable.getColumns()) {
+                    columnMap.put(column.getName().getString(), column);
+                }
+
+                for (PTable decendant : legitimateDecendants) {
+                    if (decendant.getViewType() == PTable.ViewType.READ_ONLY) {
+                        continue;
+                    }
+                    PTable newView = Transform.getTransformedView(decendant, newDataTable, columnMap, true);
+                    QueryPlan queryPlan = getQueryPlan(newView, decendant, connection);
+                    inputSplits.addAll(generateSplits(queryPlan, configuration));
+                }
```

Comment:
What happens if the views are not disjoint? Do we just harmlessly transform the same data multiple times into the same shape? Seems like we should have a test for that if we don't already.